test/controllers/update/overseas.entity.presenter.controller.spec.ts

```diff
@@ -184,5 +198,24 @@ describe("OVERSEAS ENTITY PRESENTER controller", () => {
       expect(resp.text).not.toContain(ErrorMessages.MAX_EMAIL_LENGTH);
       expect(resp.text).not.toContain(ErrorMessages.EMAIL_INVALID_FORMAT);
     });
+
+    test(`Saves transaction and OE submission with 'isRemove' flag set on POST if this is a Remove journey`, async () => {
+      const mockData = { ...APPLICATION_DATA_MOCK, [Transactionkey]: "", [OverseasEntityKey]: "" };
+      mockGetApplicationData.mockReturnValue(mockData);
+
+      const resp = await request(app).post(
+        `${OVERSEAS_ENTITY_PRESENTER_URL}${JOURNEY_REMOVE_QUERY_PARAM}`).send(PRESENTER_OBJECT_MOCK);
+
+      expect(resp.status).toEqual(302);
+
+      expect(mockData[Transactionkey]).toEqual(TRANSACTION_ID);
+      expect(mockData[OverseasEntityKey]).toEqual(OVERSEAS_ENTITY_ID);
+      expect(mockData[IsRemoveKey]).toEqual(true);
+      expect(mockTransactionService).toHaveBeenCalledTimes(1);
+      expect(mockCreateOverseasEntity).toHaveBeenCalledTimes(1);
+      expect(mockSaveAndContinue).toHaveBeenCalledTimes(1);
```

Comment:
i would add an expect to check that mockSetExtraData is called, as that is what updates the appData  in the session. You might have to declare the mockSetExtraData as I don't think it is used currently.

Reply:
👍 will do.

Reply:
New assertion added.